KenticoOnboardingApplication/test/KenticoOnboardingApplication.Contracts.Tests/Services/Wrappers/RetrievedItemTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using KenticoOnboardingApplication.Contracts.Models;
+using KenticoOnboardingApplication.Contracts.Services.Wrappers;
+using KenticoOnboardingApplication.Tests.Base.Factories;
+using NUnit.Framework;
+
+namespace KenticoOnboardingApplication.Contracts.Tests.Services.Wrappers
+{
+    [TestFixture]
+    public class RetrievedItemTests
+    {
+        private readonly Item _item = ItemsCreator.CreateItem("00000000-0000-0000-0000-000000000001", "I am  item.");
+
+        [Test]
+        public void RetrievedItem_WithNullItem_ThrowsArgumentNullException() =>
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var _ = new RetrievedItem<Item>(item: null);
+            });
+
+        [Test]
+        public void RetrievedItem_WithValidItem_ReturnsRetrievedItem()
+        {
+            var result = new RetrievedItem<Item>(_item);
+
+            Assert.That(result.Item, Is.EqualTo(_item));
+            Assert.That(result.WasFound, Is.True);
+        }
+
+        [Test]
+        public void GetItem_RetrievedItemEmpty_ThrowsInvalidOperationException()
+        {
+            var emptyRetrievedItem = RetrievedItem<Item>.Empty;
+
+            Assert.Throws<InvalidOperationException>(() =>
+            {
+                var _ = emptyRetrievedItem.Item;
```

Comment:
Love your usage of the [`_`](https://memegenerator.net/instance/48506203/mr-bean-happy-wow) here 😀